### PR TITLE
AP_Common: fix and clean up Bitmask class

### PR DESCRIPTION
**Pull request by tpwrules**
Please see commits for details.

Tested full test suite and that a vehicle seems to work okay. Saves 1.3K flash on Cube Orange.

### CODE REVIEW

File: libraries/AP_Common/Bitmask.h [hunk 1]
```diff
@@ -23,26 +23,26 @@
 
 #include <AP_InternalError/AP_InternalError.h>
 
-template<uint16_t num_bits>
+#define NUMWORDS ((NUMBITS+31)/32)
```
Comment (khancyr) [inline libraries/AP_Common/Bitmask.h:26]: isn't this require more division calculation when previously, we store do it once and store the result into numwords?

Comment (khancyr) [reply libraries/AP_Common/Bitmask.h:26]: can we change the define to a constexpr function return to get rid of the division calculation at compile time, if that isn't done by compiler already ?

**Comment by peterbarker** [comment]
Looks nice, but size-wise it could be better...

OTOH, CubeOrange is somewhere where we are going to have size problems soon...


```
Board               AP_Periph  blimp  bootloader  copter  heli   iofirmware  plane  rover  sub
Durandal                       -968   *           -1336   -1120              -1192  -1352  -1096
Hitec-Airspeed      -16                                                                    
KakuteH7-bdshot                48     *           40      40                 104    88     80
MatekF405                      -8     *           24      -48                -8     48     -8
Pixhawk1-1M-bdshot             40                 72      40                 40     64     40
f103-QiotekPeriph   -16                                                                    
f303-Universal      -16                                                                    
iomcu                                                            *                         
revo-mini                      8      *           112     56                 8      48     8
skyviper-v2450                                    48                                       
```

File: libraries/AP_Common/Bitmask.h (hunk 1 above)
Comment (tpwrules) [reply libraries/AP_Common/Bitmask.h:26]: The compiler will optimize it either way as `NUMBITS` is a constant here but it is a bit cleaner to keep it as a constant variable than a define.

**Comment by tpwrules** [comment]
We could make it a size reduction across the board by removing the new range tests in `clear` and `get`. But there is already one in `set` and having it comprehensive is IMO better for the health of the code.

Plus, having robust checks here means we can remove them elsewhere (in particular in the DroneCAN DNA server) and get some of the flash back.

**Review by tridge — APPROVED** [review]

**Pull request merged**